reactioncommerce/reaction
# [WIP] #3373 (breaking change) Convert paymentMethod.supports into an object

Working on #3373

This PR that does the following:
- Refactor payment schema with an array in `support` property
- Change the `array of String` to `object of Boolean`
- Make sure payment work properly.
- Migration of modified schema

#### Test 1
- Log in as an admin
- Open the payment settings panel
- Enable any payment method.
- Go to `Payment provider supported methods` check  or uncheck any of the supported methods the click on `save changes`. Notice the toast with a message of updated.
- Open Robomongo or it's alternative 
- Go to packages open `settings-> paymentMethodname->support` depending on the current payment method  notice the type of `Boolean` and the changed from either true to false or false to true.

#### Test 2
- Log in admin
- Open the payment settings panel
- Enable any payment method.
- Place an order and checkout.
- You will see a successful item purchase.
- Open Robomongo or it's alternative 
- Go to packages open `settings-> paymentMethodname->support` depending on the current payment method  notice the type of `Boolean` and the changed from either true to false or false to true.
